src/rest/CommunityEndpoints.js

```diff
@@ -28,87 +204,286 @@ export async function joinCommunity(communityAddress, memberAddress, secret = un
+// //////////////////////////////////////////////////////////////////
 
-export async function withdraw(communityAddress, memberAddress, wallet, confirmations = 1) {
-    const stats = await this.memberStats(communityAddress, memberAddress)
-    if (!stats.withdrawableBlockNumber) {
-        throw new Error('No earnings to withdraw.')
+/**
+ * Send a joinRequest, or get into community instantly with a community secret
+ * @param {EthereumAddress} communityAddress to join
+ * @param {String} secret (optional) if given, and correct, join the community immediately
+ */
+export async function joinCommunity(communityAddress, secret) {
+    const authKey = this.options.auth && this.options.auth.privateKey
+    if (!authKey) {
+        throw new Error('joinCommunity: StreamrClient must have auth: privateKey')
     }
-    const contract = new Contract(communityAddress, CommunityProduct.abi, wallet)
-    const withdrawTx = await contract.withdrawAll(stats.withdrawableBlockNumber, stats.withdrawableEarnings, stats.proof)
-    const tokenAddress = await contract.token()
-    const token = new Contract(tokenAddress, TestToken.abi, wallet)
-    const receipt = await withdrawTx.wait(confirmations)
-}
 
-export async function communityStats(communityAddress) {
-    const url = `${this.options.restUrl}/communities/${communityAddress}/stats`
-    return fetch(url).then((res) => res.json())
-}
+    const body = {
+        memberAddress: computeAddress(authKey)
+    }
+    if (secret) { body.secret = secret }
 
-export async function createSecret(communityAddress, secret, name = 'Untitled Community Secret') {
-    const url = `${this.options.restUrl}/communities/${communityAddress}/secrets`
+    const url = `${this.options.restUrl}/communities/${communityAddress}/joinRequests`
     return authFetch(
         url,
         this.session,
         {
             method: 'POST',
-            body: JSON.stringify({
-                name,
-                secret,
-            }),
+            body: JSON.stringify(body),
             headers: {
                 'Content-Type': 'application/json',
             },
         },
     )
 }
 
-function throwIfBadAddress(address, variableDescription) {
-    try {
-        return utils.getAddress(address)
-    } catch (e) {
-        throw new Error(`${variableDescription || 'Error'}: Bad Ethereum address ${address}`)
+/**
+ * Await this function when you want to make sure a member is accepted in the community
+ * @param {EthereumAddress} communityAddress
+ * @param {EthereumAddress} memberAddress (optional, default is StreamrClient's auth: privateKey)
+ * @param {Number} pollingIntervalMs (optional, default: 1000) ask server if member is in
+ * @param {Number} timeoutMs (optional, default: 60000) give up
+ * @return {Promise} resolves when member is in the community (or fails with HTTP error)
+ */
+export async function hasJoined(communityAddress, memberAddress, pollingIntervalMs, timeoutMs) {
```

Comment:
I feel like these should all be named parameters rather than positional i.e. `export async function hasJoined({communityAddress, memberAddress, pollingIntervalMs, timeoutMs}) {`

Reply:
also whether these are positional or not, should probably set the defaults in the parameter definition rather than at usage location i.e. `export async function hasJoined(communityAddress, memberAddress, pollingIntervalMs = 1000, timeoutMs = 60000) {`